doc/source/tutorials/emulator.rst

```diff
@@ -0,0 +1,153 @@
+
+
+.. code-block:: python
+
+    emulator_platform.instruments["pulse_simulator"].connect()
+    emulator_platform.instruments["pulse_simulator"].setup()
+    emulator_platform.instruments["pulse_simulator"].start()
+
+Likewise, for Qibolab device platforms, analogues of the following instrument methods have to be run after using the device. These methods for the pulse simulator instrument are only kept for consistency and do not have any functionality.
+
+.. code-block:: python
+
+    emulator_platform.instruments["pulse_simulator"].stop()
+    emulator_platform.instruments["pulse_simulator"].disconnect()
+
+Once the simulation is complete, it returns the same result object as any other Qibolab device platform. We can therefore retrieve these results in the same way:
+
+.. code-block:: python
+
+    results[list(results.keys())[1]].samples
+
+We can also print the simulation details.
+
+.. code-block:: python
+
+    emulator_platform.instruments["pulse_simulator"].print_sim_details()
+
+Each time a pulse sequence is simulated by the emulator, the pulse sequence and channel waveforms are added to their respective histories stored in the pulse simulator. Let us retrive the pulse sequence and channel waveform for the pulse sequence that we just executed. The full list of time steps simulated is recorded as part of the channel waveforms data.
+
+.. code-block:: python
+
+    pulse_sequence = pulse_simulator.pulse_sequence_history[-1]
+    channel_waveforms = pulse_simulator.channel_waveforms_history[-1]
+    full_time_list = channel_waveforms["time"]
+
+We can also plot the simulated waveforms of the drive channel, in this case for qubit 0 (labelled "D-0"):
+
+.. code-block:: python
+
+    import matplotlib.pyplot as plt
+
+    plt.plot(
+        channel_waveforms["time"],
+        channel_waveforms["channels"]["D-0"],
+        label="simulated waveform",
+    )
+    plt.xlabel("Time / ns")
+    plt.ylabel("Amplitude")
+    plt.legend()
+
+.. image:: emulator/simulated_waveform.png
+    :width: 500
+    :align: center
+
+Similarly, the simulation backend stores the history of states generated for each pulse sequence simulated. We can retrieve the states obtained at every time step of the simulation for the pulse sequence that we just executed as follows:
+
+.. code-block:: python
+
+    psi_t_list = simulation_backend.pulse_sim_history[-1]
+
+Finally, we can check how the fidelity of the state with all computation basis states evolves throughout the simulation to understand evolution of the qubit quantum states. In this simulation, our model also accounted for the second excited state of the qubit.
+
+.. code-block:: python
+
+    simulation_backend.fidelity_history()
+
+.. image:: emulator/fidelity_history.png
+    :width: 500
+    :align: center
```

Comment:
This is the part specific to the emulator, so I'd just focus on this one, and leave everything else up to the standard Qibolab interface (this would remove the noise added because of irrelevant implementation, as described above for `.connect()` and friends, and extract all the value of the compliance: since you implemented, you can now use it as any other Qibolab platform).

Reply:
Fixed in 6f442c1ffbb9b110c9d3118cdeb67d354c19b42a